resources/assets/lib/core-legacy/polyfills.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import DumbStorage from 'dumb-storage';
+
+declare global {
+  interface Window {
+    localStorage: Storage;
+  }
+}
+
+export default class Polyfills {
+  constructor() {
+    this.localStorage();
+  }
+
+  // Mainly for Safari Private Mode.
```

Comment:
I don't think Safari throws an error anymore in private mode 🤔

Reply:
wow they fixed things 🤔